vimeo-networking/src/main/java/com/vimeo/networking/model/ModifyVideosInAlbumSpecs.java

```diff
@@ -0,0 +1,114 @@
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An object that is used to patch video addition and deletion updates to an Album.
+ */
+@SuppressWarnings("unused")
+public class ModifyVideosInAlbumSpecs implements Serializable {
+
+    private static final long serialVersionUID = -3094719083671086785L;
+
+    @Nullable
+    @SerializedName("remove")
+    private List<NamedWrapperForRemove> mRemoveVideoList;
+
+    @Nullable
+    @SerializedName("set")
+    private List<NamedWrapperForAdd> mAddVideoList;
+
+    public ModifyVideosInAlbumSpecs(@Nullable List<RemoveVideoFromAlbum> removeVideoList,
+                                    @Nullable List<AddVideoToAlbum> addVideoList) {
+        if (removeVideoList != null) {
+            mRemoveVideoList = new ArrayList<>();
+            for (final RemoveVideoFromAlbum curRemove : removeVideoList) {
+                mRemoveVideoList.add(new NamedWrapperForRemove(curRemove));
+            }
+        }
+        if (addVideoList != null) {
+            mAddVideoList = new ArrayList<>();
+            for (final AddVideoToAlbum curAdd : addVideoList) {
+                mAddVideoList.add(new NamedWrapperForAdd(curAdd));
+            }
+        }
+    }
+
+    /**
+     * @return A nullable list of RemoveVideoFromAlbum objects, containing the wrapped list of videos to be removed
+     * from the album.
+     */
+    @Nullable
+    public List<RemoveVideoFromAlbum> getRemoveVideoList() {
+        if (mRemoveVideoList == null) {
+            return null;
+        }
+        final List<RemoveVideoFromAlbum> retVal = new ArrayList<>();
+        for (final NamedWrapperForRemove curRemove : mRemoveVideoList) {
+            retVal.add(curRemove.getRemoveVideoFromAlbum());
+        }
+        return retVal;
+    }
+
+    /**
+     * @return A nullable list of AddVideoToAlbum objects, containing the wrapped list of videos to be added
+     * to the album.
+     */
+    @Nullable
+    public List<AddVideoToAlbum> getAddVideoList() {
+        if (mAddVideoList == null) {
+            return null;
+        }
+        final List<AddVideoToAlbum> retVal = new ArrayList<>();
+        for (final NamedWrapperForAdd curAdd : mAddVideoList) {
+            retVal.add(curAdd.getAddVideoToAlbum());
+        }
+        return retVal;
+    }
+
+    private class NamedWrapperForAdd implements Serializable {
+
+        private static final long serialVersionUID = 8713902512465812810L;
+
+        @SerializedName("video")
+        private final AddVideoToAlbum mAddVideoToAlbum;
+
+        NamedWrapperForAdd(AddVideoToAlbum addVideoToAlbum) {
+            mAddVideoToAlbum = addVideoToAlbum;
+        }
+
+        /**
+         * @return The wrapped AddVideoToAlbum object.
+         */
+        AddVideoToAlbum getAddVideoToAlbum() {
+            return mAddVideoToAlbum;
+        }
+    }
+
+    private class NamedWrapperForRemove implements Serializable {
```

Comment:
They could be but that would require me to provide setters for the private members and expose private classes that it doesn't make sense to expose. In this case, I don't think we need the performance optimization since it is only used for making a patch to add videos to a user's album. Let me know if you disagree and we can talk more about it IRL.